src/docker/extract.nim

```diff
@@ -88,7 +99,49 @@ proc extractImageMark(self: ChalkObj): string =
   finally:
     discard runDockerGetEverything(@["rm", containerId])
 
-proc extractMarkFromSigStore(self: ChalkObj): string =
+proc extractMarkFromLayer(self: ChalkObj): string =
+  var err = "no chalk mark found in any of the registry images"
+  for image in self.images.uniq():
+    let spec = image.withDigest(self.imageDigest)
+    trace("docker: extract chalk mark from registry for " & $spec)
+    try:
+      let manifest = fetchImageManifest(spec, platform = self.platform)
+      if not manifest.hasChalkLayer():
+        raise newException(
+          ValueError,
+          "Last layer for " & $spec & " does not contain /chalk.json",
+        )
+      let lastLayer = manifest.layers[^1]
+      return lastLayer.nameRef.layerGetFSFileString(
+        name   = "chalk.json",
+        accept = manifest.mediaType,
+      )
```

Comment:
I prefer  pattern to exit early on error so that the rest of the function doesnt need to be indented as much

```
for i in foo():
  if i.bad():
    continue
  i.good()
```

I especially prefer this when the `good()` logic is more involved where its a dozens of lines. having everything one more indent I think doesnt help, especially when splitting buffers where the left real-estate is more precious